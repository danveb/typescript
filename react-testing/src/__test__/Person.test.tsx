import { render, screen } from "@testing-library/react";
import { Person } from "../Person";

describe("test Person component", () => {
  // smoke test
  test("renders without crashing", () => {
    render(<Person name="Danny" />); 
  });

  // snapshot test
  test("creates a snapshot", () => {
    const { asFragment } = render(
      <Person name="Jack" />
    ); 
    expect(asFragment()).toMatchSnapshot(); 
  }); 

  // getByText
  test("renders a name", () => {
    render(<Person name="Sophie" />); 
    const text = screen.getByText("Name is Sophie"); 
    expect(text).toBeInTheDocument(); 
  });

  // getByRole
  test("finds aria-role", () => {
    render(<Person name="Juan" />); 
    const divElement = screen.getByRole("contentinfo"); 
    expect(divElement).toHaveTextContent("Name is Juan"); 
    expect(divElement).toBeInTheDocument(); 
    expect(divElement).toHaveAttribute("role", "contentinfo"); 
  }); 
});