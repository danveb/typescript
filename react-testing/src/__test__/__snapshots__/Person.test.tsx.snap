// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Person component creates a snapshot 1`] = `
<DocumentFragment>
  <div
    role="contentinfo"
  >
    Name is Jack
  </div>
</DocumentFragment>
`;
